src/io/flutter/run/SdkRunConfig.java

```diff
@@ -84,11 +98,56 @@ public LaunchState getState(@NotNull Executor executor, @NotNull ExecutionEnviro
     final Project project = env.getProject();
     final RunMode mode = RunMode.fromEnv(env);
     final Module module = ModuleUtil.findModuleForFile(mainFile.getFile(), env.getProject());
-
     final LaunchState.Callback callback = (device) -> {
       if (device == null) return null;
 
-      final GeneralCommandLine command = fields.createFlutterSdkRunCommand(project, device, mode);
+      final GeneralCommandLine command = getCommand(env, device);
+      {
+        // Workaround for https://github.com/flutter/flutter/issues/16766
+        // TODO(jacobr): remove once flutter tool incremental building works
+        // properly with --track-widget-creation.
+        final Path buildPath = command.getWorkDirectory().toPath().resolve("build");
+        final Path cachedParametersPath = buildPath.resolve("last_build_run.json");
+        final String[] parametersToTrack = {"--preview-dart-2", "--track-widget-creation"};
+        final JsonArray jsonArray = new JsonArray();
+        for (String parameter : command.getParametersList().getList()) {
+          for (String allowedParameter : parametersToTrack) {
+            if (parameter.startsWith(allowedParameter)) {
+              jsonArray.add(new JsonPrimitive(parameter));
+              break;
+            }
+          }
+        }
+        String json = new Gson().toJson(jsonArray);
+        String existingJson = null;
+        if (Files.exists(cachedParametersPath)) {
+          try {
+            existingJson = new String(Files.readAllBytes(cachedParametersPath), StandardCharsets.UTF_8);
+          }
+          catch (IOException e) {
+            LOG.warn("Unable to get existing json from " + cachedParametersPath);
+          }
+        }
+        if (!json.equals(existingJson)) {
+          // We don't have proof the current run is consistent with the existing run.
```

Comment:
Here's a simpler, less elegant solution:

If the launch args contain `--track-widget-creation`, then always delete `build/snapshot_blob.bin.d.fingerprint` before a run.

Thoughts?

Reply:
unfortunately that doesn't quite work as there is also a problem currently that if you run with the flag and then without the flag you will have a bad combination of widgets tracking and not tracking locations due to over eager caching